src/index.js

```diff
@@ -210,13 +216,22 @@ function shimBelongsTo(target) {
         return original.apply(this, arguments);
       }
 
-      let foreignKeyValue = instance.get(this.foreignKey);
       return Promise.resolve().then(() => {
+        const foreignKeyValue = instance.get(this.foreignKey);
         if (foreignKeyValue === undefined || foreignKeyValue === null) {
           return Promise.resolve(null);
         }
 
-        const loader = options[EXPECTED_OPTIONS_KEY].loaders[this.target.name].bySingleAttribute[this.targetKey];
+        const loaders = options[EXPECTED_OPTIONS_KEY].loaders;
+        let loader = loaders[this.target.name].bySingleAttribute[this.targetKey];
```

Comment:
Fixing the paranoid issue for `belongsTo` relationship, basically using the same idea from `shimModel` function.